tests.py

```diff
@@ -0,0 +1,127 @@
+import pytest
+import pickle
+from bs4 import BeautifulSoup
+from unittest import mock
+
+import scrape.category as category
+import scrape.offer as offer
+import scrape.utils as utils
+
+REGIONS_TO_TEST = ["Gdań", "Sop", "Oliw", "Wrzeszcz", "czechowice", "Nowa Wieś", "pomorskie", "Książąt pomor sopot", ""]
+ACTUAL_REGIONS = [{"city": "gdansk_40"}, {"city": "sopot_208"}, {"[district_id]": 51316, "city": "gdansk_40"}, {"[district_id]": 30,
+                  "city": "gdansk_40"}, {"city": "czechowice-dziedzice_2258"}, {"city": "nowa-wies_6001"}, {"region": "pomorskie"}, {"[street_id]": 15544, "city": "sopot_208"}, {}]
+
+@pytest.mark.parametrize(
+    'text,dic,expected_value', [
+        ('ala', {'a': 'b'}, 'blb'),
+        ('Gdańsk', {'ń': 'n'}, 'Gdansk')
+     ])
+def test_replace_all(text, dic, expected_value):
+    assert utils.replace_all(text, dic) == expected_value
+
+
+@pytest.mark.parametrize('region_part,expected_value', zip(REGIONS_TO_TEST, ACTUAL_REGIONS))
+def test_get_region_from_autosuggest(region_part, expected_value):
+    assert utils.get_region_from_autosuggest(region_part) == expected_value
+
+
+# @pytest.mark.parametrize("main_category", ["wynajem", "sprzedaz"])
+# @pytest.mark.parametrize("detail_category", ["mieszkanie", "dom", "pokoj", "dzialka", "lokal", "haleimagazyny", "garaz", ""])
+# @pytest.mark.parametrize("get_region_from_autosuggest_value", ACTUAL_REGIONS)
+# @pytest.mark.parametrize("region", REGIONS_TO_TEST)
+# def test_get_url(main_category, detail_category, region, get_region_from_autosuggest_value):
+#         with mock.patch("scrape.utils.get_region_from_autosuggest") as get_region_from_autosuggest:
+#             get_region_from_autosuggest.return_value = get_region_from_autosuggest_value
+#             url = utils.get_url(main_category, detail_category, region)
+
+@pytest.mark.parametrize(
+    'markup_file,expected_value', [
+        ("test_data/markup_no_offers", "95c3ad18f9c716209fdfc5d73b13f4a64fd12fc7ca9b6d0d4a5f60ce80b574b3"),
+        ("test_data/markup_offers", "d6e9f6202c0fd68ddc539a54bd728d59aa27d7276470818a57ed7c3c2db5f612")
+    ])
+def test_get_csrf_token(markup_file, expected_value):
+    with open(markup_file, "rb") as file:
```

Comment:
file is builtin name use sth else,
maybe `markup_file` could be `markup_path` and then as `markup_file`?